### PR TITLE
Fix broken links and Add awesome_bot linkcheck 

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

Fixes some broken links and Adds Checks for links in Markdown Files

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://aboutcode.readthedocs.io/en/latest/scancode-toolkit/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -89,6 +89,17 @@ matrix:
       python: "3.6"
       dist: bionic
 
+    - os: linux
+      sudo: required
+      before_install:
+        - gem install awesome_bot
+      script:
+        - awesome_bot .github/*.md --allow-redirect --allow-dupe --skip-save-results
+        - awesome_bot .github/ISSUE_TEMPLATE/*.md --allow-redirect --allow-dupe --skip-save-results
```
Comment (AyanSinhaMahapatra) [inline .travis.yml]: These check for broken links inside .md files used for GitHub templates. Caught [here](https://github.com/nexB/scancode-toolkit/commit/d82929ccd197cad0b83697575cc42027dead47b2#r37224758).

File: .travis.yml [hunk 2]
```diff
@@ -89,6 +89,17 @@ matrix:
...
+        - awesome_bot *.rst --allow-redirect --allow-dupe --skip-save-results
```
Comment (AyanSinhaMahapatra) [inline .travis.yml]: These check for broken links inside .rst files at Project Root.

Files - AUTHORS.rst, CHANGELOG.rst, CODE_OF_CONDUCT.rst, CONTRIBUTING.rst, README.rst.

File: .travis.yml [hunk 3]
```diff
@@ -89,6 +89,17 @@ matrix:
...
+        - awesome_bot */**/*.md --allow-redirect --allow-dupe --skip-save-results
+        - awesome_bot */*.ABOUT --allow-redirect --allow-dupe --skip-save-results
```
Comment (AyanSinhaMahapatra) [inline .travis.yml]: These check for broken links inside Thirdparty .ABOUT files and plugins/plugins-builtin README.md files.

**Comment by AyanSinhaMahapatra** [comment]
Basically uses (awesome_bot)[https://github.com/dkhamsing/awesome_bot] to look for broken links in some parts of the Project. Other parts can be added obviously and this can be moved inside a script to clear .travis.yml, will add these changes if suggested.

To look for Broken Links offline, refer (Installation)[https://github.com/dkhamsing/awesome_bot#installation], and to make sense of the cli refer output (examples)[https://github.com/dkhamsing/awesome_bot#installation] or (command-line)[https://github.com/dkhamsing/awesome_bot#command-line].

**Comment by AyanSinhaMahapatra** [comment]
See this Successful [Travis Job](https://travis-ci.org/nexB/scancode-toolkit/jobs/650929032?utm_medium=notification&utm_source=github_status) to view the scans.

File: .travis.yml (hunk 1 above)
Comment (pombredanne) [reply .travis.yml]: You could put that directly as comments in the yml. Use # at the start of a line for commenting, like in Python

Comment (AyanSinhaMahapatra) [reply .travis.yml]: @pombredanne Done!

**Pull request merged**